src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultIEnumerableConverter.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal class JsonIEnumerableT<TSource> : IEnumerable<TSource>
+    {
+        private List<TSource> _list;
+        public JsonIEnumerableT(IList sourceList)
+        {
+            _list = new List<TSource>();
```

Comment:
Since `IList<T>` implements `IEnumerable<T>` we should just be able to set `_list` directly and not have to convert (same for other types)

Reply:
However there is a concern that for nested lists (lists of lists) - these would be able to be cast these to `IList` for example for an `IEnumerable`-only.

For performance we could add an internal interface like `IListConverter<T>` (or base class) that we implement on these converter types. It has a `Add` method. We detect this interface (like we do with `IList`) and just populate it directly. Along with this we would also change the type on the PropertyInfo class from `IEnumerable` (for example) to `JsonIEnumerableT` so the type is just auto-created like with other concrete collections.

Reply:
Per our offline discussion, we are going to utilize the `EnumerableConverter` construct to populate and return an instance of  a `JsonEnumerableT` class that implements the supported interface collections.